pallets/nfts/src/lib.rs

```diff
@@ -233,25 +269,49 @@ impl<T: Config> NFTs for Pallet<T> {
     type AccountId = T::AccountId;
     type NFTDetails = T::NFTDetails;
     type NFTId = T::NFTId;
+    type NFTSeriesId = T::NFTSeriesId;
 
     fn create(
         owner: &Self::AccountId,
         details: Self::NFTDetails,
+        series_id: Self::NFTSeriesId,
     ) -> result::Result<Self::NFTId, DispatchError> {
-        let id = Total::<T>::get();
-        Total::<T>::put(id.checked_add(&1.into()).ok_or(Error::<T>::NFTIdOverflow)?);
+        // Check if the owner is even allowed to add anything to the series.
+        if series_id != Default::default() {
+            let series_owner = Series::<T>::get(series_id).0;
+            ensure!(
+                series_owner == Default::default() || series_owner == *owner,
+                Error::<T>::NotSeriesOwner
+            );
+        }
+
+        // Create the nft.
+        let nft_id = Total::<T>::get();
+        Total::<T>::put(
+            nft_id
+                .checked_add(&1.into())
+                .ok_or(Error::<T>::NFTIdOverflow)?,
+        );
+
+        // Create or expand a series.
+        if series_id != Default::default() {
+            Self::add_nft_to_series(owner.clone(), nft_id, series_id)?;
```

Comment:
You are `get`ing the serie before, I'd suggest modifying this function call to avoid reading the same value twice

Reply:
The whole thing should now be simpler, easier and it should consume less reads/writes.